### ISSUE TITLE
Disable anonymous user

### ISSUE THREAD

**Issue by MaGaoju** [goharbor/harbor#10760]
we are using Harbor internally and using OIDC provider for the auth. we don't want anonymous access for this instance. we'd like to keep all public project only accessible by authed users. it's possible to make this configurable ?   if code change needed, anyone can help to share this changes? 

thx!

**Comment by cafeliker** [comment]
@reasonerjt I am reading the Harbor codes, could you pls help check if we can remove this section to disable the anonymous access?
![image](https://user-images.githubusercontent.com/6274204/74898190-1f359980-53d4-11ea-98fe-870f7051f7fc.png)

https://github.com/goharbor/harbor/blob/v1.10.1/src/core/api/project.go#L406

Thanks
ye

**Comment by steven-zou** [comment]
@xaleeks 

Obviously, this is a requirement and then needs the decision from PM.

**Comment by cafeliker** [comment]
Anonymous access is good for community purpose; and for Enterprise, the public repo should be only visible for authenticated people to protect the content and IP - so pls add the configuration item to enable/disable anonymous.

Before it's available, can we quickly fork and make some changes as I proposed above? Do you think it's the only place I need to change?

Thank you again for such great open source software.

ye

**Comment by renmaosheng** [comment]
I got a similar request from "Xing ye Yin hang".

**Comment by cafeliker** [comment]
I submitted a PR https://github.com/goharbor/harbor/pull/10825.

Thanks
ye

**Comment by xaleeks** [comment]
agreed, makes sense for a lot of customers

**Comment by reasonerjt** [comment]
I agree it's a valid requirement but we should put it in the pipeline and write proposal, then target it to a release, no?

**Comment by cafeliker** [comment]
I will make the document, thanks for your information.

**Comment by cafeliker** [comment]
@xaleeks I created the document with a PR in the harbor community repo at https://github.com/goharbor/community/pull/124.

Thanks
ye

**Comment by xaleeks** [comment]
@cafeliker Thanks, I think it looks good.

**Comment by sdileepkumarreddy** [comment]
Hi Any update in this feature? Would be nice to have this for handling some analytics

**Comment by shreekanthDev** [comment]
Hello Team, any update on this feature implementation?
